src/parser/command.rs

```diff
@@ -205,17 +205,22 @@ pub(crate) fn parse_defcal_gate<'a>(
 pub(crate) fn parse_defcal_measure<'a>(
     input: ParserInput<'a>,
 ) -> InternalParserResult<'a, Instruction> {
-    let (input, qubit_index) = opt(token!(Integer(v)))(input)?;
-    let qubit = qubit_index.map(Qubit::Fixed);
-    let (input, destination) = token!(Identifier(v))(input)?;
+    let (input, (qubit, destination)) = map(
+        tuple((opt(parse_qubit), opt(token!(Identifier(v))))),
+        |(a, b)| match (a, b) {
+            (Some(qubit), Some(destination)) => (Some(qubit), destination),
+            (Some(destination), None) => (None, destination.to_string()),
+            _ => panic!("Implement this error"),
+        },
+    )(input)?;
```

Comment:
This is my first time really trying to parse something non-trivial in quil-rs. Is this acceptable robust? All existing tests pass, plus a new one for the issue this code solves. I don't love the `panic!` but I went with it because thats how all the errors inside of a `map` are implemented elsewhere.

Reply:
Looks like you could use [`map_res`](https://docs.rs/nom/latest/nom/combinator/fn.map_res.html) ?

```rs
    let (input, (qubit, destination)) = map_res(
        tuple((opt(parse_qubit), opt(token!(Identifier(v))))),
        |(a, b)| match (a, b) {
            (Some(qubit), Some(destination)) => Ok((Some(qubit), destination)),
            (Some(destination), None) => Ok((None, destination.to_string())),
            _ => Err(InternalError::from_kind(
                input,
                ParserErrorKind::UnexpectedEOF("No DEFCAL target"),
            )),
        },
    )(input)?;
```

Reply:
Although, since the first token is not optional:

```rs
    let (input, (qubit, destination)) = map(
        tuple((parse_qubit, opt(token!(Identifier(v))))),
        |(a, b)| match (a, b) {
            (qubit, Some(destination)) => (Some(qubit), destination),
            (destination, None) => (None, destination.to_string()),
        },
    )(input)?;
```

Reply:
Oh, that last one is clever, nice thinking.